07-performance.Rmd

```diff
@@ -133,20 +133,17 @@ local(source("code/07-performance_f2b.R", local=TRUE))
 
 ###  Converting factors to numerics {-}
 
-<!-- __XXX__Colin__ Occassionally below: on what occassions? Curious -->
-
 A factor is just a vector of integers with associated levels. Occasionally we want to
-convert a factor into its numerical equivalent. The most efficient way of doing this
-(especially for long factors) is
+convert a factor into its numerical equivalent. For example, our data 
+has been converted to factor by `read.csv` since it contained a string. Using
 
 ```{r, echo=2, results="hide"}
 f = factor(10:100)
```

Comment:
I don't understand this example. To me it seems unrealistic having a factor of numerical data. Much more commonly factors are used for character data, e.g.

```r
f = factor(rep(letters, 20))
as.numeric(levels(f))[f]
```

But the second line of that example fails (producing all NAs), making me wonder how I'd use it in practice. `as.numeric(as.character(f))` also fails there. How about

```r
f = factor(rep(letters, 20))
unique(as.numeric(f))
```

As a more realistic example that actually contains characters as factors do in real life?